repository/avail-light.repository.ts

```diff
@@ -48,8 +49,9 @@ export async function runLC(onBlock: Function, registerUnsubscribe: Function): P
         //Create required info for process block
         const block: Block = { number: blockNumber, hash: blockHash, totalCellCount: totalCellCount, confidence: 0, sampleCount: sampleCount, timestamp }
         const matrix: Matrix = { maxRow: r, maxCol: c, verifiedCells: [], totalCellCount }
-        onBlock(block, matrix, randomCells, proofs, commitments)
+        onBlock((list: BlockToProcess[]) => [...list, { block, matrix, randomCells, proofs, commitments }])
+        //onBlock(block, matrix, randomCells, proofs, commitments)
     });
 
     return registerUnsubscribe(() => unsubscribe);
```

Comment:
This should only return the unsubscribe method, instead of registering it on the apps state from here.

Reply:
Yes, I agree with this. We should not change the state from outside the app; I will change this.